deployments/container/Makefile

```diff
@@ -55,18 +54,48 @@ endif
 
 DOCKERFILE = $(CURDIR)/deployments/container/Dockerfile
 
-# Build the container image
+DOCKER_BUILD_CACHE_OPTIONS :=
+
+ifeq ($(CACHE_TO_GHCR), true)
+  DOCKER_BUILD_CACHE_OPTIONS +=	\
+	--cache-to type=registry,ref=ghcr.io/nvidia/k8s-dra-driver-gpu:buildcache,mode=max
```

Comment:
Question: How does this handle concurrent builds? (that is to say if multiple PRs are updating the same cache?)

Reply:
Great question. Thanks for looking!!

That's fine. There are only atomic flips involved.

We can say with certainty that if there is a cache _hit_ then the cached layer can be used.

When doing vastly different builds concurrently however the value of this caching method becomes small: we have a moving target here defined with `:buildcache`. I opted for this method to not have the cache grow large. The last update wins, so to say. The way I understand this, it yields most benefit across builds when the builds are highly similar (have layers in common). And when the upload/download dance is faster than building locally, of course.

There may of course also be unnecessary upload if multiple runners upload the same layer 'at the same time' (and only one of them wins).